### PR TITLE
doc: matter: change link to Matter spec

### PR DESCRIPTION
**Pull request by greg-fer**
Updated links to Matter specification download.
Removed notes about restricted access.
Updated related Zigbee links.
Updated naming for consistency with the Matter page. KRKNWK-16070.

Signed-off-by: Grzegorz Ferenc <Grzegorz.Ferenc@nordicsemi.no>

### CODE REVIEW

**Comment by greg-fer** [comment]
@Damian-Nordic , @kkasperczyk-no , can you kindly verify that the changes in Matter spec names are accurate? I would not want to mention a Core spec where Device Library spec is due, and so on.

**Review by sebastiandraus — APPROVED** [review]

**Review by ArekBalysNordic — APPROVED** [review]

**Review by LuDuda — APPROVED** [review]
Looks good!

Can we also mention the link to the specification already in this page: `ug_matter_intro_overview.rst` ?

**Review by doublemis1 — APPROVED** [review]

File: doc/nrf/ug_matter_gs_testing_thread_separate_otbr_linux_macos.rst [hunk 1]
```diff
@@ -42,6 +42,15 @@ Configuring the environment
 
 To configure and use Thread Border Router and CHIP Tool for Linux or macOS on separate devices, complete the following steps.
 
+.. note::
+      This setup is showcased in the "Developing Matter 1.0 products with nRF Connect SDK", embedded below.
+
+      .. raw:: html
+
+         <div style="text-align: center;">
+         <iframe width="560" height="315" src="https://www.youtube.com/embed/9Ar13rMxGIk?t=805" title="Developing Matter 1.0 products with nRF Connect SDK webinar video" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
```
Comment (LuDuda) [inline doc/nrf/ug_matter_gs_testing_thread_separate_otbr_linux_macos.rst:51]: Does this link work? When i manually enter this link it does not jump to the specified time?

File: doc/nrf/ug_matter_overview_multi_fabrics.rst [hunk 2]
```diff
@@ -12,6 +12,15 @@ Because each Matter fabric has an administrator, this allows the device to be ad
 
 This Matter feature is key for achieving Matter's smart home interoperability.
 
+.. note::
```
Comment (LuDuda) [inline doc/nrf/ug_matter_overview_multi_fabrics.rst:15]: It's very subjective.. but personally i would prefer to have youtube video embedded only on main documentation page and if anything, link to the webinar/main page from here. I have a feeling we might have too many videos embedded on different pages, and videos tends to become outdated.

File: doc/nrf/ug_matter_gs_testing_thread_separate_otbr_linux_macos.rst (hunk 1 above)
Comment (greg-fer) [reply doc/nrf/ug_matter_gs_testing_thread_separate_otbr_linux_macos.rst]: It does for me.

File: doc/nrf/ug_matter_overview_multi_fabrics.rst (hunk 2 above)
Comment (greg-fer) [reply doc/nrf/ug_matter_overview_multi_fabrics.rst]: Let's discuss as part of another task: https://projecttools.nordicsemi.no/jira/browse/KRKNWK-16086 I removed these additions from this PR.

**Pull request merged**